### PR TITLE
5 packages from benfaerber/liquid-ml at 0.1

### PR DESCRIPTION
**Pull request by benfaerber**
This pull-request concerns:
- `liquid_interpreter.0.1`: The interpreter for Liquid
- `liquid_ml.0.1`: Shopify's Liquid templating language in OCaml
- `liquid_parser.0.1`: The parser for Liquid
- `liquid_std.0.1`: The Standard Libarary for Liquid
- `liquid_syntax.0.1`: The Syntax Definitions for Liquid

---
* Homepage: https://github.com/benfaerber/liquid-ml
* Source repo: git+https://github.com/benfaerber/liquid-ml.git
* Bug tracker: https://github.com/benfaerber/liquid-ml/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/liquid_interpreter/liquid_interpreter.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+description: """
+The interpreter for Liquid
+"""
+maintainer: "faerberbendev@protonmail.com"
+authors: ["Ben Faerber"]
+homepage: "https://github.com/benfaerber/liquid-ml"
+bug-reports: "https://github.com/benfaerber/liquid-ml/issues"
+dev-repo: "git+https://github.com/benfaerber/liquid-ml.git"
+license: "MIT"
+depends: [ "base" "core" "stdio" "re2" "liquid_syntax" "liquid_parser" "liquid_std" ]
```
Comment (raphael-proust) [inline packages/liquid_interpreter/liquid_interpreter.0.1/opam:12]: You need to add some dependencies and some constraints.

First to ocaml:
```
"ocaml" { >= "4.11" }
```

This is based on the results of the CI. You get failures for some of the versions pre-4.11.

Second to `dune`
```
"dune" { >= 2.5 }
```

The version number is based on your `dune-project` file.

Finally, for constraints of packages that are part of the same group
```
"liquid_syntax" { = version }
```

This ensures that the packages are installed with the same version. This is the recommended way to deal with packages that are cut from the same repository.

-----------

In summary

```suggestion
depends: [
  "ocaml" { >= "4.11" }
  "dune" { >= "2.5" }
  "base"
  "core"
  "stdio"
  "re2"
  "liquid_syntax" { = version }
  "liquid_parser" { = version }
  "liquid_std" { = version }
]
```

And similarly for the other packages too.

There are probably version constraints needed for the other dependencies (base, core, stdio, re2) but I don't know what they are.

Comment (raphael-proust) [reply packages/liquid_interpreter/liquid_interpreter.0.1/opam:12]: For the constraints on base, core, stdio, re2, batteries, calendar, etc. you can refer to https://github.com/ocaml/opam-repository/wiki/FAQ which has explanations on how to find lower-bound constraints.

**Comment by benfaerber** [comment]
I've made the requested changes. Thanks for the help!

File: packages/liquid_interpreter/liquid_interpreter.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+depends: [ 
+    "ocaml" { >= "4.11" }
+    "dune" { >= "2.5" } 
+    "base" { >= "0.13.1" }
+    "core" { >= "0.13.0" } 
+    "stdio" { >= "0.13.0" } 
+    "re2" { >= "0.13.0" }
+    "liquid_syntax" { = "0.1"} 
+    "liquid_parser" { = "0.1"}
+    "liquid_std" { = "0.1"}
```
Comment (raphael-proust) [inline packages/liquid_interpreter/liquid_interpreter.0.1/opam:21]:
```suggestion
    "liquid_syntax" { = version} 
    "liquid_parser" { = version}
    "liquid_std" { = version}
```

It doesn't change much inside the opam-repository itself, but it's probably a good idea to use `version` upstream so you don't have to change it when you make a new release.

`version` is a variable (or a keyword I'm not sure) which holds (or expands to or whatever) the version of the package file that it is in. So `version` inside `packages/liquid_interpreter/liquid_interpreter.0.1/opam` is the same value as `"0.1"`, but `version` inside `packages/liquid_interpreter/liquid_interpreter.0.2/opam` will be the same value as `"0.2"`.

File: packages/liquid_ml/liquid_ml.0.1/opam [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "ocaml" { >= "4.11" }
+    "dune" { >= "2.5" } 
+    "base" { >= "0.13.1" }
+    "core" { >= "0.13.0" } 
+    "stdio" { >= "0.13.0" } 
+    "re2" { >= "0.13.0" }
+    "liquid_syntax" { = "0.1"} 
+    "liquid_parser" { = "0.1"}
+    "liquid_std" { = "0.1"}
+    "liquid_interpreter" { = "0.1"}
```
Comment (raphael-proust) [inline packages/liquid_ml/liquid_ml.0.1/opam:23]:
```suggestion
    "liquid_syntax" { = version} 
    "liquid_parser" { = version}
    "liquid_std" { = version}
    "liquid_interpreter" { = version}
```

File: packages/liquid_parser/liquid_parser.0.1/opam [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+license: "MIT"
+
+depends: [ 
+    "ocaml" { >= "4.11" }
+    "dune" { >= "2.5" } 
+    "base" { >= "0.13.1" }
+    "core" { >= "0.13.0" } 
+    "stdio" { >= "0.13.0" } 
+    "re2" { >= "0.13.0" }
+    "liquid_syntax" { = "0.1"}
```
Comment (raphael-proust) [inline packages/liquid_parser/liquid_parser.0.1/opam:20]:
```suggestion
    "liquid_syntax" { = version} 
```

File: packages/liquid_std/liquid_std.0.1/opam [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "dune" { >= "2.5" } 
+    "base" { >= "0.13.1" }
+    "core" { >= "0.13.0" } 
+    "stdio" { >= "0.13.0" } 
+    "re2" { >= "0.13.0" }
+    "base64" { >= "2.0.0" }
+    "sha" { >= "1.0" }
+    "liquid_syntax" { = "0.1"} 
+    "liquid_parser" { = "0.1"}
+    "liquid_syntax" { = "0.1"}
```
Comment (raphael-proust) [inline packages/liquid_std/liquid_std.0.1/opam:23]:
```suggestion
    "liquid_syntax" { = version} 
    "liquid_parser" { = version}
    "liquid_syntax" { = version} 
```

Comment (benfaerber) [reply packages/liquid_std/liquid_std.0.1/opam:23]: Oh I see! I thought you meant for me to change to my version. I will apply the change

**Comment by raphael-proust** [comment]
Considering https://github.com/ocaml/opam-repository/pull/24334, is it still relevant to merge this PR here?

**Comment by benfaerber** [comment]
> Considering #24334, is it still relevant to merge this PR here?

No, I will close it here. Thanks
